### ISSUE TITLE
If post content contains a shortcode with WP_Query the edit post page will receive wrong data

### ISSUE THREAD

**Issue by ktmn** [Automattic/amp-wp#1105]
Twenty Seventeen 1.5
AMP 0.7-RC1-4edea76-20180423T183146Z

Create this shortcode as a test:

```php
add_shortcode('some_query', function() {
	$output = '';

	$the_query = new \WP_Query(['post_type' => 'post']);
	if($the_query->have_posts()) {
		$output .= '<ul>';
		while($the_query->have_posts()) {
			$the_query->the_post();
			$output .= '<li>' . get_the_title() . '</li>';
		}
		$output .= '</ul>';
		wp_reset_postdata();
	}

	return $output;
});
```

It does the most basic WP_Query as specified in the [usage docs](https://codex.wordpress.org/Class_Reference/WP_Query).

Now add it to any post's content `[some_query /]` and save the post. 
When you go to edit that post again, the shortcode is evaluated in the back end, I suppose for the validation purposes that are added in 0.7. 
But even though the shortcode calls `wp_reset_postdata()` it seems the main query post is set to one from the shortcode's query results and therefore the post title and content is incorrect.

**Comment by westonruter** [comment]
Wow. Good find. I can reproduce the issue.

**Comment by westonruter** [comment]
@ktmn please give this a try: https://github.com/Automattic/amp-wp/pull/1107

**Comment by westonruter** [comment]
Fix has been merged into the `0.7` branch.
